elements/pf-select/pf-select.ts

```diff
@@ -0,0 +1,517 @@
+import { LitElement, html, type PropertyValues } from 'lit';
+import { customElement } from 'lit/decorators/custom-element.js';
+import { property } from 'lit/decorators/property.js';
+import { query } from 'lit/decorators/query.js';
+import { styleMap } from 'lit/directives/style-map.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { type ListboxValue } from '@patternfly/pfe-core/controllers/listbox-controller.js';
+import { ToggleController } from '@patternfly/pfe-core/controllers/toggle-controller.js';
+import type { PfSelectOption } from './pf-select-option.js';
+import { PfChipGroup } from '@patternfly/elements/pf-chip/pf-chip-group.js';
+import { PfSelectList } from './pf-select-list.js';
+
+import styles from './pf-select.css';
+
+export type PfSelectItemsDisplay = '' | 'badge' | 'chips';
+export type Placement = 'bottom' | 'top' | 'top-start' | 'top-end' | 'bottom-start' | 'bottom-end';
+/**
+ * A select list enables users to select one or more items from a list.
+ *
+ * A select component consists of a toggle control to open and close a menu of actions or links.
+ * Selects differ from dropdowns in that they persist selection,
+ * whereas dropdowns are typically used to present a list of actions or links.
+ *
+ * @slot - insert `pf-select-option` and/or `pf-select-groups` here
+ */
+@customElement('pf-select')
+export class PfSelect extends LitElement {
+  static readonly styles = [styles];
+
+  /**
+   * whether listbox is always expanded
+   */
+  @property({ attribute: 'always-expanded', type: Boolean }) alwaysExpanded = false;
+
+  /**
+   * whether filtering (if enabled) will be case-sensitive
+   */
+  @property({ attribute: 'case-sensitive', type: Boolean }) caseSensitive = false;
+
+  /**
+   * text for a special option that allows user to create an option from typeahead input text;
+   * set to '' in order to disable this feature
+   */
+  @property({ attribute: 'create-option-text', type: String }) createOptionText = '';
+
+  /**
+   * ARIA label for chip group used to describe chips
+   */
+  @property({ attribute: 'current-selections-label', type: String }) currentSelectionsLabel = 'Current selections';
+
+  /**
+   * listbox button text when single-select listbox has no selected option text
+   */
+  @property({ attribute: 'default-text', type: String }) defaultText = 'Options';
+
+  /**
+   * whether select is disabled
+   */
+  @property({ reflect: true, attribute: 'disabled', type: Boolean }) disabled = false;
+
+  /**
+   * whether option filtering is disabled
+   */
+  @property({ reflect: true, attribute: 'disable-filter', type: Boolean }) disableFilter = false;
+
+  /**
+   * enable to flip listbox when it reaches boundary
+   */
+  @property({ attribute: 'enable-flip', type: Boolean }) enableFlip = false;
+
+  /**
+   * whether listbox is has checkboxes when `multi-select` is enabled
+   */
+  @property({ reflect: true, attribute: 'has-checkboxes', type: Boolean }) hasCheckboxes = false;
```

Comment:
should this be a private getter, or context protocol? don't let the user say "has checkboxes, promise!", instead, just check if it has checkboxes and note that state privately.

Reply:
@bennypowers this is actually a prop that the consumer can set. Consumer can choose to do multiselect with or without checkboxes.